operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/EventDispatcherTest.java

```diff
@@ -141,7 +147,7 @@ void callDeleteOnControllerIfMarkedForDeletionWhenNoFinalizerIsConfigured() {
     eventDispatcher.handleExecution(
         executionScopeWithCREvent(Watcher.Action.MODIFIED, testCustomResource));
 
-    verify(controller).deleteResource(eq(testCustomResource), any());
+    verify(controller, never()).deleteResource(eq(testCustomResource), any());
```

Comment:
This isn't correct. Delete should be called.

Reply:
Yes @metacosm , this is the test case where we are ignoring finalizer configuration. I modified the `configureToNotUseFinalizer()` method as below mentioned setup was missing when `configureToNotUseFinalizer()` creates a new `configuration`.

```
    when(configuration.getName()).thenReturn("EventDispatcherTestController");
    when(configService.getMetrics()).thenReturn(Metrics.NOOP);
    when(configuration.getConfigurationService()).thenReturn(configService);
```

I updated the PR, please let me know if anything is missing here.